### DISCUSSION TITLE
Mosquitto Docker Compose v2

### DISCUSSION THREAD

**Discussion by vvatelot** [vvatelot/mosquitto-docker-compose#10] [Announcements]
Hello to all, some time ago, @spagnoloG, @nixilb and @martinbra reported [a bug in the use of this project](https://github.com/vvatelot/mosquitto-docker-compose/issues/1). It took me a while to find out and I apologize. Thanks to them for insisting and proposing fixes!

So we upgraded the project from [v1.0.0 to v2.0.0](https://github.com/vvatelot/mosquitto-docker-compose/releases/tag/v2.0.0). v1.0.0 uses mosquitto 1.6 while v2.0.0 uses mosquitto 2.0. We also took advantage of this to change the structure of local volumes. To migrate from version 1 to version 2, here are the steps to follow:

- Modify the access rights to the log and config directories: `sudo chown -R $UID:$GID log config`
- When you start your container, use the command `UID=$UID GID=$GID docker-compose up -d`

**Comment by jpmiller25** [comment]
Can I use environment variables UID and PID inside the docker compose definition instead of defining them when I run docker compose up? I tried but it seems it keeps trying to chown the volumes to user 1883.

**Reply by vvatelot** [reply]
Hello @jpmiller25, there might be a better and cleaner way to do so. For now, it is the best I found out to work with UID and GID. If you find something better, do not hesitate to propose it :)
